File: Components/eXpressApp Framework/FeatureCenter/CS/FeatureCenter.Module.Web/Updater.cs
using System;

using DevExpress.Xpo;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.ExpressApp.Xpo;

using FeatureCenter.Module.Web.PropertyEditors;
using FeatureCenter.Module.PropertyEditors;
using FeatureCenter.Module.Web.Layout;

namespace FeatureCenter.Module.Web
{
    public class Updater : ModuleUpdater
    {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema()
        {
            base.UpdateDatabaseAfterUpdateSchema();

			if(ObjectSpace is XPObjectSpace) {
				CreateStringInSpecialFormatPropertiesObjects();
				CreateLayoutItemTemplatesObjects();
			}

            ObjectSpace.CommitChanges();
        }

		private void CreateStringInSpecialFormatPropertiesObjects() {
			if(ObjectSpace.FindObject<WebStringInSpecialFormatPropertiesObject>(null) == null) {
				WebStringInSpecialFormatPropertiesObject stringPropertiesObject = ObjectSpace.CreateObject<WebStringInSpecialFormatPropertiesObject>();
				stringPropertiesObject.Name = "Object 1";
				stringPropertiesObject.HtmlStringProperty = "<h1>String Property</h1><p>with <font color=\"red\" size=\"5\">HTML</font> <strong>formatting</strong></p>";
				stringPropertiesObject.Save();
			}
		}
		private void CreateLayoutItemTemplatesObjects() {
            if(ObjectSpace.FindObject<ItemTemplates>(null) == null) {
				ItemTemplates itemTemplates = ObjectSpace.CreateObject<ItemTemplates>();
				itemTemplates.Name = "Object 1";
				itemTemplates.Save();
			}
		}
	}
}

File: Components/eXpressApp Framework/MainDemo/VB/MainDemo.Module.Web/DatabaseUpdate/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Persistent.Base.Security
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp.PivotChart.Web
Imports DevExpress.ExpressApp.PivotChart
Imports MainDemo.Module.DatabaseUpdate

Namespace MainDemo.Module.Web.DatabaseUpdate
    Public Class Updater
        Inherits ModuleUpdater
        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            CType(New TaskAnalysis1LayoutUpdater(), TaskAnalysis1LayoutUpdater).Update(ObjectSpace.FindObject(Of Analysis)(CriteriaOperator.Parse("Name='Completed tasks'")))
            CType(New TaskAnalysis2LayoutUpdater(), TaskAnalysis2LayoutUpdater).Update(ObjectSpace.FindObject(Of Analysis)(CriteriaOperator.Parse("Name='Estimated and actual work comparison'")))
        End Sub
    End Class
    Public Class TaskAnalysis1LayoutUpdater
        Inherits TaskAnalysis1LayoutUpdaterBase
        Protected Overrides Function CreateAnalysisControl() As IAnalysisControl
            Return New AnalysisControlWeb()
        End Function
        Protected Overrides Function CreatePivotGridSettingsStore(ByVal control As IAnalysisControl) As DevExpress.Persistent.Base.IPivotGridSettingsStore
            Return New ASPxPivotGridSettingsStore((CType(control, AnalysisControlWeb)).PivotGrid)
        End Function
    End Class
    Public Class TaskAnalysis2LayoutUpdater
        Inherits TaskAnalysis2LayoutUpdaterBase
        Protected Overrides Function CreateAnalysisControl() As IAnalysisControl
            Return New AnalysisControlWeb()
        End Function
        Protected Overrides Function CreatePivotGridSettingsStore(ByVal control As IAnalysisControl) As DevExpress.Persistent.Base.IPivotGridSettingsStore
            Return New ASPxPivotGridSettingsStore((CType(control, AnalysisControlWeb)).PivotGrid)
        End Function
    End Class
End Namespace
